util-core/src/main/scala/com/twitter/util/Future.scala

```diff
@@ -1404,7 +1422,7 @@ def join[%s](%s): Future[(%s)] = join(Seq(%s)).map { _ => (%s) }""".format(
     sizeThreshold: Int,
     timeThreshold: Duration = Duration.Top,
     sizePercentile: => Float = 1.0f
-  )(f: Seq[In] => Future[Seq[Out]]
+  )(f: Iterable[In] => Future[Seq[Out]]
```

Comment:
This is a breaking change, just want to make sure this is necessary. If so we should add an entry to CHANGELOG(i can add it on my branch).

Reply:
Maybe it's better to keep this at scala.collection.Seq then.

* For 2.13 it *may* be a breaking change, but some refactoring is already expected there.
* For 2.12.x it won't break anything.

I will investigate, and come back to it this Tuesday, as I'm heading into a long weekend.

Reply:
https://github.com/twitter/util/pull/248/commits/b5b124bddac097e1044e68a29c266964ac5a16b1 is a way to do it without the Iterable

Reply:
The tests seem to be mocking me over that change.

I've never used any mocking frameworks. What's the correct way to fix this? Include function composition in the mock?

Reply:
Hey @martijnhoekstra, I modified tests to mock the composed method, and verify against `m`. 
```
val m = mock[Any => Future[Seq[Int]]]
val f = new Predef.Function[Seq[Int], Future[Seq[Int]]] {
  override def apply(v1: Seq[Int]): Future[Seq[Int]] = ???
  override def compose[A](g: A => Seq[Int]): A => Future[Seq[Int]] = m
}
```

I have your branch internally almost ready, passed all tests but only running the rest of benchmarks.